tx-pool/src/component/orphan.rs

```diff
@@ -78,6 +78,11 @@ impl OrphanPool {
                 self.by_out_point
                     .get_mut(&out_point)
                     .map(|set| set.remove(id));
+
+                // remove empty by_out_point entry
```

Comment:
I think you should test `self.by_out_point[out_point]` is empty.

Reply:
Yes, I made a mistake.